### PR TITLE
Refactor Test Framework API Surface.

### PR DESCRIPTION
**Pull request by ozevren**

### CODE REVIEW

File: galley/pkg/testing/testdata/dataset/extensions/v1beta1/ingress_merge_0_expected.json [hunk 1]
```diff
@@ -74,7 +74,7 @@
             "route": [
               {
                 "destination": {
-                  "host": "service2.ns.svc.cluster.local",
+                  "host": "service2.{{.Namespace}}.svc.cluster.local",
```
Comment (nmittler) [inline galley/pkg/testing/testdata/dataset/extensions/v1beta1/ingress_merge_0_expected.json]: the second part (`.svc.cluster.local`) is configured as the `Domain`.  Not sure if this is something you can add here as well.

File: pkg/test/util/yml/tracker.go [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+	"io/ioutil"
+	"os"
+	"path"
+	"strings"
+	"sync"
+)
+
+// Tracker tracks the life-cycle of Yaml based resources. It stores single-part Yaml files on disk and updates the
+// files as needed, as the resources change.
+type Tracker struct {
```
Comment (nmittler) [inline pkg/test/util/yml/tracker.go]: nit: would a better name be `cache`?

File: galley/pkg/testing/testdata/dataset/extensions/v1beta1/ingress_merge_0_expected.json (hunk 1 above)
Comment (ozevren) [reply galley/pkg/testing/testdata/dataset/extensions/v1beta1/ingress_merge_0_expected.json]: I want to focus on Galley tests and converting them into compliance suite later. I think that something we must do, but I'd rather do a more focused PR for that purpsoe.

**Pull request closed without merging**